improver_tests/calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -387,59 +308,26 @@ def setUp(self):
     def test_coefficients_from_mean(self):
         """Test that the expected coefficient cube is returned when the
         ensemble mean is used as the predictor."""
-        expected_coeff_names = ["gamma", "delta", "alpha", "beta"]
-        result = self.plugin.create_coefficients_cube(
+        coeff_names = ["gamma", "delta", "alpha", "beta"]
+        expected_coeff_names = [f"emos_coefficient_{s}" for s in coeff_names]
+        result = self.plugin.create_coefficients_cubelist(
             self.optimised_coeffs, self.historic_forecast
         )
         self.assertEqual(result, self.expected)
-        self.assertEqual(self.plugin.coeff_names, expected_coeff_names)
+        self.assertEqual([cube.name() for cube in result], expected_coeff_names)
 
     @ManageWarnings(ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
     def test_coefficients_from_realizations(self):
         """Test that the expected coefficient cube is returned when the
         ensemble realizations are used as the predictor."""
-        expected_coeff_names = ["gamma", "delta", "alpha", "beta0", "beta1", "beta2"]
+        coeff_names = ["gamma", "delta", "alpha", "beta0", "beta1", "beta2"]
```

Comment:
Ah OK.  I see the use case.  But I wonder if this is the best design?  I would think that having a single "beta" cube with a realization coordinate would be a better way of storing this information.

Happy to discuss if you'd like to.

Reply:
I've now restructured the beta coefficient cube as suggested.